lib/core/src/Cardano/Wallet/StakePool/Metrics.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedLabels #-}
+-- | This module can fold over a blockchain to collect metrics about
+-- Stake pools.
+--
+-- It interacts with:
+-- - "Cardano.Wallet.Network" which provides the chain
+-- - "Cardano.Wallet.DB" - which can persist the metrics
+-- - "Cardano.Wallet.Api.Server" - which presents the results in an endpoint
+module Cardano.Wallet.StakePool.Metrics where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader (..), PoolId (..), SlotId (..) )
+import Data.Map.Strict
+    ( Map )
+import Data.Word
+    ( Word )
+import Fmt
+import GHC.Generics
+    ( Generic )
+
+import qualified Data.Map.Strict as Map
+
+data State = State
+    { tip :: BlockHeader
+    , numberOfBlocksProducedThisEpoch :: Map PoolId Word
```

Comment:
I think this should be some kind of `Map Epoch (Map PoolId Word)`, now, to make rollback easier, and if we will get access to stake-distributions of past epoch.